modules/dnn/test/test_layers_1d.cpp

```diff
@@ -25,124 +51,787 @@ TEST_P(Layer_1d_Test, Scale)
     lp.set("bias_term", false);
     Ptr<ScaleLayer> layer = ScaleLayer::create(lp);
 
-    std::vector<int> input_shape = {batch_size, 3};
-    std::vector<int> output_shape = {batch_size, 3};
-
-    if (batch_size == 0){
-        input_shape.erase(input_shape.begin());
-        output_shape.erase(output_shape.begin());
-    }
-
-    cv::Mat input = cv::Mat(input_shape, CV_32F, 1.0);
+    input = Mat(dims, input_shape.data(), CV_32F);
     cv::randn(input, 0.0, 1.0);
-    cv::Mat weight = cv::Mat(output_shape, CV_32F, 2.0);
+    Mat weight = Mat(dims, output_shape.data(), CV_32F, 2.0);
 
     std::vector<Mat> inputs{input, weight};
-    std::vector<Mat> outputs;
+    Mat output_ref = input.mul(weight);
 
-    cv::Mat output_ref = input.mul(weight);
-    runLayer(layer, inputs, outputs);
+    TestLayer(layer, inputs, output_ref);
+}
 
-    ASSERT_EQ(shape(output_ref), shape(outputs[0]));
-    normAssert(output_ref, outputs[0]);
+TEST_P(Layer_Test, ReLU6)
+{
+    LayerParams lp;
+
+    lp.type = "ReLU6";
+    lp.name = "ReLU6Layer";
+    lp.set("min_value", 0.0);
+    lp.set("max_value", 1.0);
+    Ptr<ReLU6Layer> layer = ReLU6Layer::create(lp);
+
+    Mat output_ref(dims, output_shape.data(), CV_32F, 1.0);
+    std::vector<Mat> inputs{input};
+
+    TestLayer(layer, inputs, output_ref);
+}
+
+TEST_P(Layer_Test, Clip)
+{
+    LayerParams lp;
+
+    lp.type = "Clip";
+    lp.name = "clipLayer";
+    lp.set("min_value", 0.0);
+    lp.set("max_value", 1.0);
+    Ptr<ReLU6Layer> layer = ReLU6Layer::create(lp);
```

Comment:
There is no difference between `Layer_Test.ReLU6` and `Layer_Test.Clip`

Reply:
I left it explicitly, since it is not obvious that ReLU6 function as clips our frame work. lets leave it